test/client/query-subscribe.js

```diff
@@ -11,596 +12,828 @@ module.exports = function(options) {
+        }
+      }
+      if (total === 10) {
+      // first document is its own batch; then subsequent creates
+      // are debounced until after all other 9 docs are created
+        expect(batchSizes).eql([1, 9]);
+        done();
+      }
+    });
+
+    // create an initial document. this will lead to the 'insert'
+    // event firing the first time, while sharedb is definitely
+    // debouncing
+    connection.get('items', '0').on('error', done).create({});
+  });
+
+  it('db.pollDebounce option reduces subsequent poll interval', function(done) {
+    var connection = this.backend.connect();
+    this.backend.db.canPollDoc = function() {
+      return false;
+    };
+    this.backend.db.pollDebounce = 1000;
+    var query = connection.createSubscribeQuery('items', this.matchAllDbQuery);
+    query.on('error', done);
+    var batchSizes = [];
+    var total = 0;
+
+    query.on('insert', function(docs) {
+      batchSizes.push(docs.length);
+      total += docs.length;
+      if (total === 1) {
+      // first write received by client. we're debouncing. create 9
+      // more documents.
+        for (var i = 1; i < 10; i++) {
+          connection.get('items', i.toString()).on('error', done).create({});
+        }
+      }
+      if (total === 10) {
+      // first document is its own batch; then subsequent creates
+      // are debounced until after all other 9 docs are created
+        expect(batchSizes).eql([1, 9]);
+        done();
+      }
+    });
+
+    // create an initial document. this will lead to the 'insert'
+    // event firing the first time, while sharedb is definitely
+    // debouncing
+    connection.get('items', '0').on('error', done).create({});
+  });
+
+  it('pollInterval updates a subscribed query after an unpublished create', function(done) {
+    var connection = this.backend.connect();
+    this.backend.suppressPublish = true;
+    var query = connection.createSubscribeQuery('dogs', this.matchAllDbQuery, {pollInterval: 50}, function(err) {
+      if (err) return done(err);
+      connection.get('dogs', 'fido').on('error', done).create({});
+    });
+    query.on('error', done);
+    query.on('insert', function(docs) {
+      expect(util.pluck(docs, 'id')).eql(['fido']);
+      done();
+    });
+  });
+
+  it('db.pollInterval updates a subscribed query after an unpublished create', function(done) {
+    var connection = this.backend.connect();
+    this.backend.suppressPublish = true;
+    this.backend.db.pollInterval = 50;
+    var query = connection.createSubscribeQuery('dogs', this.matchAllDbQuery, null, function(err) {
+      if (err) return done(err);
+      connection.get('dogs', 'fido').on('error', done).create({});
+    });
+    query.on('error', done);
+    query.on('insert', function(docs) {
+      expect(util.pluck(docs, 'id')).eql(['fido']);
+      done();
+    });
+  });
+
+  it('pollInterval captures additional unpublished creates', function(done) {
+    this.timeout(4000);
```

Comment:
Let's use sinon fake timers here, to avoid artificially lengthening the test case.

Will have to advance time at least 3 times, we think?

Reply:
Added fakeTimers